src/commands/project/deploy/report.ts

```diff
@@ -51,23 +57,77 @@ export default class DeployMetadataReport extends SfCommand<DeployResultJson> {
       summary: messages.getMessage('flags.results-dir.summary'),
       helpGroup: testFlags,
     }),
+    // we want to allow undefined for a simple check deploy status
+    // eslint-disable-next-line sf-plugin/flag-min-max-default
+    wait: Flags.duration({
+      char: 'w',
+      summary: deployMessages.getMessage('flags.wait.summary'),
+      description: deployMessages.getMessage('flags.wait.description'),
+      unit: 'minutes',
+      helpValue: '<minutes>',
+      min: 1,
+    }),
   };
 
   public async run(): Promise<DeployResultJson> {
     const [{ flags }, cache] = await Promise.all([this.parse(DeployMetadataReport), DeployCache.create()]);
-    const jobId = cache.resolveLatest(flags['use-most-recent'], flags['job-id']);
+    const jobId = cache.resolveLatest(flags['use-most-recent'], flags['job-id'], false);
+
+    const deployOpts = cache.get(jobId) ?? ({} as DeployOptions & { isMdapi: boolean });
+    const waitDuration = flags['wait'];
+    const org = flags['target-org'] ?? (await Org.create({ aliasOrUsername: deployOpts['target-org'] }));
 
-    const deployOpts = cache.get(jobId);
-    const org = await Org.create({ aliasOrUsername: deployOpts['target-org'] });
-    const [deployStatus, componentSet] = await Promise.all([
-      // we'll use whatever the org supports since we can't specify the org
+    // if we're using mdapi we won't have a component set
+    let componentSet = new ComponentSet();
+    if (!deployOpts.isMdapi) {
```

Comment:
the isMapi is such a large big codePath that I think it would be a good to have a NUT that
1. does a deploy with --metadata-dir from outside a project
2. does a report about that, using the cache
3. deos a report about that with the cache deleted